dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -317,8 +316,8 @@ RtpsUdpDataLink::associated(const RepoId& local_id, const RepoId& remote_id,
     RtpsWriter& w = writers_[local_id];
     w.remote_readers_[remote_id].durable_ = remote_durable;
     w.durable_ = local_durable;
-    enable_heartbeat = true;
-
+    w.ready_to_hb_ = !(local_durable && remote_durable);
```

Comment:
Simplify this expression: there is no way for a matched remote reader to be durable when the local writer isn't

Reply:
Sure. I originally had this is as "w.ready_to_hb_ = !local_durable" which was insufficient in the case when the remote reader wasn't durable so I just added remote_durable to the condition. Since the matched remote reader can't be durable when the local writer isn't durable, then I guess you can just use "!remote_durable" to cover all cases when the durable data doesn't need to be added before the first heartbeat.